tests/types/tinheritance_arrays.nim

```diff
@@ -0,0 +1,53 @@
+discard """
```

Comment:
@beef331 is the intention with these tests to fold into the spec?

Reply:
It probably should be moved to the spec I guess since it's for regression assurance.

Reply:
OK, if you know how to do it now then go for it, otherwise ask @haxscramper when he's about and do it later?